packages/mdc-list/foundation.ts

```diff
@@ -657,33 +712,41 @@ export class MDCListFoundation extends MDCFoundation<MDCListAdapter> {
    * toggleCheckbox value. User interaction should not toggle list item(s) when
    * disabled.
    */
-  private setSelectedIndexOnAction(index: number, toggleCheckbox = true) {
+  private setSelectedIndexOnAction(index: number, isCheckboxAlreadyUpdatedInAdapter: boolean) {
     if (this.isCheckboxList) {
-      this.toggleCheckboxAtIndex(index, toggleCheckbox);
+      this.toggleCheckboxAtIndex(index, isCheckboxAlreadyUpdatedInAdapter);
+      this.adapter.notifySelectionChange([index]);
     } else {
-      this.setSelectedIndex(index);
+      this.setSelectedIndex(index, {isUserInteraction: true});
     }
   }
 
-  private toggleCheckboxAtIndex(index: number, toggleCheckbox: boolean) {
+  private toggleCheckboxAtIndex(index: number, isCheckboxAlreadyUpdatedInAdapter: boolean) {
     const selectionAttribute = this.getSelectionAttribute();
-    let isChecked = this.adapter.isCheckboxCheckedAtIndex(index);
-
-    if (toggleCheckbox) {
-      isChecked = !isChecked;
-      this.adapter.setCheckedCheckboxOrRadioAtIndex(index, isChecked);
+    const adapterIsChecked = this.adapter.isCheckboxCheckedAtIndex(index);
+
+    // By default the checked value from the adapter is toggled unless the
+    // checked state in the adapter has already been updated beforehand.
+    // This can be happen when the underlying native checkbox has already
```

Comment:
Hm, I'm not sure this is supposed to be possible --  I believe we disable pointer-events on all contained inputs (in css). Is there a situation in which the underlying control might still be toggled directly?

Reply:
Yeah, good point. In Angular Material IIRC we disabled this so that we would not need this special logic, but based on the existance of this property (and the way it is used in the component adapter), it seemed like this was still needed, especially since any arbitrary `input[type="checkbox/radio]` would be picked up by the `MDCList` component.

If this is no longer needed (which would be nice), then we'd need to double-check/enforce that whether pointer events are actually prevented + we need to enforce that the native control is not part of the tab order (as one could still tab to the input and press space right?). maybe that should be a follow-up? I just cleaned this up a little.

Reply:
In general I think the goal was to allow for the native `change` event to work, but that seems to not work correclty anyway wtihin an list item.

Reply:
All excellent points. I think most of this is legacy, unfortunately -- the pointer-events stuff came with List Evolution. Before that, the controls were clickable (though list would manipulate tabindex IIRC). 

The latest guidance is for list items to be the sole focus of interaction; any embedded control is, at least in spirit if not implementation, a visual indicator. 

All that being said, I think addressing this is out of scope. So, as long as we're able to preserve existing behavior (while adding the new event), we should be in good shape. Is this consistent with the current state of the PR?

Reply:
@brandondiamond thanks for the confirmation. I agree that the embedded control is really just a visual indicator. Let me know if you'd want some help on addressing this in the future (happy to help!)

For this PR: correct. I really just renamed the property from `toggleCheckbox` to something a little more accurate w/ some additional comments. Mostly because I found it confusing and very inaccurate. Everything should behave the same, just the `notifySelectionChange` event is added on top of the existing logic.